bundles/org.openhab.persistence.influxdb/src/main/java/org/openhab/persistence/influxdb/internal/InfluxDBMetadataUtils.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.persistence.influxdb.internal;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.items.Metadata;
+import org.openhab.core.items.MetadataKey;
+import org.openhab.core.items.MetadataRegistry;
+import org.openhab.persistence.influxdb.InfluxDBPersistenceService;
+
+/**
+ * Logic to use items metadata from an openHAB {@link Item}
+ *
+ * @author Johannes Ott - Initial contribution
+ */
+@NonNullByDefault
+public class InfluxDBMetadataUtils {
+
+    private InfluxDBMetadataUtils() {
+    }
+
+    public static String calculateMeasurementNameFromMetadata(InfluxDBConfiguration configuration,
+            final @Nullable MetadataRegistry currentMetadataRegistry, String name, @Nullable String itemName) {
+
+        if (itemName == null || currentMetadataRegistry == null) {
+            return name;
+        }
+
+        if (configuration.isUseMetadataMeasurementName()) {
```

Comment:
Okay will change this!

Reply:
While writing documentation about usage of metadata for additional information, I wonder now whether this configuration option is needed at all. User will configure it's item like this:

```
Number testItem1 {... influxdb= "" [tag1=foo, tag2= bar]}
or
Number testItem2 {... influxdb= "measurement_name" [tag1=foo, tag2=bar]}
```
Thinking again about this, it could be handled without this additional configuration? What do you think @lujop and @Hilbrand?